lib/DataFileUtil/DataFileUtilImpl.py

```diff
@@ -1082,6 +1354,109 @@ def versions(self, ctx):
         # return the results
         return [wsver, shockver]
 
+    def download_staging_file(self, ctx, params):
+        """
+        Download a staging area file to scratch area
+        :param params: instance of type "DownloadStagingFileParams" (Input
+           parameters for the "download_staging_file" function. Required
+           parameters: staging_file_subdir_path: subdirectory file path e.g.
+           for file: /data/bulk/user_name/file_name staging_file_subdir_path
+           is file_name for file:
+           /data/bulk/user_name/subdir_1/subdir_2/file_name
+           staging_file_subdir_path is subdir_1/subdir_2/file_name) ->
+           structure: parameter "staging_file_subdir_path" of String
+        :returns: instance of type "DownloadStagingFileOutput" (Results from
+           the download_staging_file function. copy_file_path: copied file
+           scratch area path) -> structure: parameter "copy_file_path" of
+           String
+        """
+        # ctx is the context object
+        # return variables are: results
+        #BEGIN download_staging_file
+        if not params.get('staging_file_subdir_path'):
+          error_msg = "missing 'staging_file_subdir_path' parameter"
+          raise ValueError(error_msg)
+        
+        staging_file_subdir_path = params.get('staging_file_subdir_path')
+        staging_file_name = os.path.basename(staging_file_subdir_path)
+        token_user = ctx['user_id']
+        staging_file_path = self._get_staging_file_path(
+            token_user, staging_file_subdir_path)
+
+        self.log('Start downloading staging file: %s' % staging_file_path)
+        dstdir = os.path.join(self.scratch, 'tmp')
```

Comment:
This could also potentially suffer from overwrite problems if multiple DFUs are running at once, or it other modules use the tmp dir